Makefile

```diff
@@ -103,7 +122,12 @@ vet: ## Run go vet against code.
 
 .PHONY: test
 test: manifests generate fmt vet envtest ## Run tests.
```

Comment:
Does `make test` work on s390x?

I'm asking because, IIUC, `test` depends on `envtest` which will attempt to install `setup-envtest` . And `setup-envtest` is not available for s390x, it is?

Reply:
It installs `setup-envtest` which is available for s390x.

Reply:
The test has been passed with the changes (the binaries' versions to 1.24.2): http://jenkins.katacontainers.io/job/confidential-containers-operator-main-ubuntu-20.04-s390x-containerd_kata-qemu-PR/30/. It includes the following:

```
test -s /home/jenkins/workspace/confidential-containers-operator-main-ubuntu-20.04-s390x-containerd_kata-qemu-PR/bin/setup-envtest || GOBIN=/home/jenkins/workspace/confidential-containers-operator-main-ubuntu-20.04-s390x-containerd_kata-qemu-PR/bin go install sigs.k8s.io/controller-runtime/tools/setup-envtest@latest
go: downloading sigs.k8s.io/controller-runtime/tools/setup-envtest v0.0.0-20221214170741-69f093833822
go: downloading sigs.k8s.io/controller-runtime v0.14.0
go: downloading github.com/spf13/afero v1.6.0
KUBEBUILDER_ASSETS="${PWD}/kubebuilder-tools/bin" go test ./... -coverprofile cover.out
?   	github.com/confidential-containers/operator	[no test files]
?   	github.com/confidential-containers/operator/api/v1beta1	[no test files]
ok  	github.com/confidential-containers/operator/controllers	10.702s	coverage: 0.0% of statements
```

Reply:
Great. I managed to reserve a s390x machine and I can confirm `make test` works.

BTW, I tried other targets like `make build` and I had to export `CC=gcc`. Non-s390x expert speaking: should we export CC on Makefile as well?

Reply:
Oh, I haven’t tried that out yet;; Thanks for letting me know. Yes, it would be great. I will reflect that right now.

Reply:
I've succeeded building a binary without passing `CC=gcc`:

```
root@a314lp42:~/test/operator# make build
mkdir -p /root/test/operator/bin
test -s /root/test/operator/bin/controller-gen || GOBIN=/root/test/operator/bin go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.9.2                                                                 
go: downloading sigs.k8s.io/controller-tools v0.9.2
go: downloading github.com/spf13/cobra v1.4.0
go: downloading github.com/fatih/color v1.12.0
go: downloading golang.org/x/tools v0.1.10-0.20220218145154-897bd77cd717
go: downloading k8s.io/api v0.24.0
go: downloading gopkg.in/yaml.v2 v2.4.0
go: downloading k8s.io/apimachinery v0.24.0
go: downloading gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
go: downloading k8s.io/apiextensions-apiserver v0.24.0
go: downloading sigs.k8s.io/yaml v1.3.0
go: downloading github.com/gobuffalo/flect v0.2.5
go: downloading github.com/mattn/go-colorable v0.1.8
go: downloading github.com/mattn/go-isatty v0.0.12
go: downloading github.com/gogo/protobuf v1.3.2
go: downloading github.com/google/gofuzz v1.1.0
go: downloading k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
go: downloading k8s.io/klog/v2 v2.60.1
go: downloading gopkg.in/inf.v0 v0.9.1
go: downloading sigs.k8s.io/structured-merge-diff/v4 v4.2.1
go: downloading golang.org/x/net v0.0.0-20220127200216-cd36cc0744dd
go: downloading sigs.k8s.io/json v0.0.0-20211208200746-9f7c6b3444d2
go: downloading golang.org/x/sys v0.0.0-20220209214540-3681064d5158
go: downloading github.com/json-iterator/go v1.1.12
go: downloading github.com/modern-go/reflect2 v1.0.2
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading golang.org/x/xerrors v0.0.0-20200804184101-5ec99f83aff1
go: downloading golang.org/x/mod v0.6.0-dev.0.20220106191415-9b9b3d81d5e3
/root/test/operator/bin/controller-gen object:headerFile="hack/boilerplate.go.txt" paths="./..."
go fmt ./...
go vet ./...
go: downloading github.com/onsi/ginkgo v1.16.5
go: downloading github.com/onsi/gomega v1.19.0
go: downloading github.com/nxadm/tail v1.4.8
go: downloading gopkg.in/tomb.v1 v1.0.0-20141024135613-dd632973f1e7
go build -o bin/manager main.go
```

Reply:
Hmm... interesting. There should be some setup missing on  my environment. For the records, I've RHEL 8.5 installed and I am using Golang 1.18.9 that I download from go.dev.

Reply:
oh, I have tested it on Ubuntu 20.04.